synapse/lib/cell.py

```diff
@@ -2628,7 +2653,18 @@ async def addHttpsPort(self, port, host='0.0.0.0', sslctx=None):
         }
         serv = self.wapp.listen(port, address=addr, ssl_options=sslctx, **kwargs)
         self.httpds.append(serv)
-        return list(serv._sockets.values())[0].getsockname()
+
+        lhost, lport = list(serv._sockets.values())[0].getsockname()
+
+        if managed_ctx:
+            # If the Cell is managing the SSLCtx, then we register a reload handler
+            # for it which reloads the pkey / cert.
+            def reload():
+                sslctx.load_cert_chain(certpath, pkeypath)
+
+            self.addReloadFunc(f'HTTPS_SSL_Reload_port={lport}', reload)
```

Comment:
Maybe add something about certs in this name? At first, I thought it was reloading the listening port, not the certificates.

Reply:
Updated